lib/hard-module.js

```diff
@@ -56,7 +61,7 @@ function needRebuild(buildTimestamp, fileDependencies, contextDependencies, file
     if(!ts) timestamp = Infinity;
     if(ts > timestamp) timestamp = ts;
   });
-  return timestamp >= buildTimestamp;
+  return timestamp >= buildTimestamp && needsMd5Rebuild;
 }
 
 HardModule.needRebuild = needRebuild;
```

Comment:
`needRebuild` is used in watch mode. Basically webpack creates a module through the factory, checks if a module with the same identifier is in the memory cache, and calls `needRebuild` on the cache. Since webpack is doing that call and not hard source we can't have webpack pass these new arguments.

HardSource reuses the function, since the idea is the same. In the case of HardSource we can work with some extra info on boot. Having the md5 info during later watch mode builds would help throw out false negatives but not help with false positives like this change for the boot time `needRebuild` check.